utils/build-icons/lib/processIcon.js

```diff
@@ -1,57 +1,24 @@
 'use strict';
 
-const { messages, builder, buildSVGDataURI } = require('./processIconHelpers');
+const { messages, builder, buildSVG } = require('./processIconHelpers');
 const { iconSize } = require('./config');
 
-/* Parse the XML. 
-  * Warn if there is an defs (because that means an i and a xlink which could collide).
-  *
-  * Once we are sure icons from design are single paths, we can extract the
-  * path here as a root element. for now, warn if there are nodes that
-  * are not paths or if there is a group or if there is more then one path.
+/* Parse the XML.
+ * We can add various warning and analysis checks here, but for now,
+ * we are primarily concerned about sizing of the viewbox.
 */
-const processIcon = ({ icon, fileName, id }) => {
+const processIcon = ({ icon, fileName, id, svgIds }) => {
   const innerContent = icon.svg;
-
-  if (innerContent['def']) {
-    return console.log(messages.def(fileName));
-  }
-
-  if (innerContent['g']) {
-    console.log(messages.g(fileName));
-  }
-
-  if (Object.keys(innerContent).length != 2) {
-    return console.log(messages.multiCombined(fileName));
-  }
+  const result = { id, fileName, svgIds };
 
   if (!innerContent['$'].viewBox) {
-    return console.log(messages.missingViewBox(fileName));
+    return { ...result, error: messages.missingViewBox(fileName) };
   } else if (innerContent['$'].viewBox != `0 0 ${iconSize} ${iconSize}`) {
-    return console.log(messages.viewBoxSize(fileName, iconSize));
+    return { ...result, error: messages.viewBoxSize(fileName, iconSize) };
   }
 
-  if (innerContent.path) {
-    if (innerContent.path.length > 1) {
-      return console.log(messages.multiRootPaths(fileName));
-    }
-    const path = builder.buildObject({ path: innerContent.path[0] });
-    const viewBox = innerContent['$'].viewBox;
-    const xmlns = innerContent['$'].xmlns;
-
-    return { id, svgDataUri: buildSVGDataURI(xmlns, viewBox, path) };
-  } else if (innerContent.g) {
-    if (innerContent.g.length > 1) {
-      return console.log(messages.multiRootPaths(fileName));
-    }
-    const path = builder.buildObject({ g: innerContent.g[0] });
-    const viewBox = innerContent['$'].viewBox;
-    const xmlns = innerContent['$'].xmlns;
-
-    return { id, svgDataUri: buildSVGDataURI(xmlns, viewBox, path) };
-  } else {
-    return console.log(messages.missingTopLevelPathOrGroup(fileName));
-  }
```

Comment:
I can't remember from when I actually spent some time with this stuff – are any of the current assets being handled by the other cases here? I feel like there were a couple.

Either way I thought it was pretty nice to have some of the edge cases handled. It might be nice to err on the side of caution ahead of time (in this case) so designers can throw a little more at it before needing to involve an engineer.

Reply:
None of these "edge cases" should apply any more, because we now have moved id / path stuff to the designer's responsibility. In other words, we only used to care about this because we were trying to 

a) ultimately generate a single path instead of a full SVG
b) warn if there were ids because we were removing all of them

However, now we don't remove ids, and we instead require them to be unique. And since we are writing out the entire SVG, we don't care about multiple groups or paths.

Reply:
Ah, okay. I think I misunderstood the purpose of the original code to some degree. 
Makes sense 👌